lib/ansible/template/__init__.py

```diff
@@ -198,6 +217,11 @@ def is_template(data, jinja_env):
     comment = False
     d2 = jinja_env.preprocess(data)
 
+    # Quick check to see if this is remotely like a template before doing
+    # more expensive investigation.
+    if not is_possibly_template(data, jinja_env):
```

Comment:
I think we should be using `d2` instead of `data` here.  Or move this before `found = None` even.  I have no idea what the performance hit is of `Environment.preprocess`.

Reply:
Oh, woops. I meant to do that.